dart/lib/src/default_integrations.dart

```diff
@@ -13,22 +13,59 @@ class RunZonedGuardedIntegration extends Integration {
 
   final Future<void> Function() _runner;
 
+  /// Needed to check if we somehow caused a `print()` recursion
+  bool _isPrinting = false;
+
   @override
   FutureOr<void> call(Hub hub, SentryOptions options) {
-    runZonedGuarded(() async {
-      await _runner();
-    }, (exception, stackTrace) async {
-      // runZonedGuarded doesn't crash the App.
-      final mechanism = Mechanism(type: 'runZonedGuarded', handled: true);
-      final throwableMechanism = ThrowableMechanism(mechanism, exception);
-
-      final event = SentryEvent(
-        throwable: throwableMechanism,
-        level: SentryLevel.fatal,
-      );
-
-      await hub.captureEvent(event, stackTrace: stackTrace);
-    });
+    runZonedGuarded(
+      () async {
+        await _runner();
+      },
+      (exception, stackTrace) async {
+        // runZonedGuarded doesn't crash the App.
+        final mechanism = Mechanism(type: 'runZonedGuarded', handled: true);
+        final throwableMechanism = ThrowableMechanism(mechanism, exception);
+
+        final event = SentryEvent(
+          throwable: throwableMechanism,
+          level: SentryLevel.fatal,
+        );
+
+        await hub.captureEvent(event, stackTrace: stackTrace);
+      },
+      zoneSpecification: ZoneSpecification(
+        print: (self, parent, zone, line) {
+          if (!hub.isEnabled || !options.enablePrintBreadcrumbs) {
```

Comment:
```suggestion
          if (!options.enablePrintBreadcrumbs || !hub.isEnabled) {
```
its cheaper to check the boolean first, I guess